src/bootstrap/test.rs

```diff
@@ -1126,7 +1126,19 @@ note: if you're sure you want to do this, please open an issue as to why. In the
                 Ok(path) => path,
                 Err(_) => p,
             })
-            .filter(|p| p.starts_with(suite_path) && (p.is_dir() || p.is_file()))
+            .filter(|p| p.starts_with(suite_path))
+            .filter(|p| {
+                let exists = p.is_dir() || p.is_file();
+                if !exists {
+                    if let Some(p) = p.to_str() {
+                        builder.info(&format!(
+                            "Warning: Skipping \"{}\": file or directory does not exist",
```

Comment:
Hm, it looks like a path can reference something that's not a file or a directory (e.g. broken symlink) - maybe we should say something like "is not a regular file or director" here instead?

Reply:
Sounds fair, updated